src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -190,14 +194,18 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     {
                         return;
                     }
-
-                    IEnumerable<ConfigurationClient> availableClients = _configClientManager.GetAvailableClients(utcNow);
-
-                    if (!availableClients.Any())
-                    {
-                        _logger.LogDebug(LogHelper.BuildRefreshSkippedNoClientAvailableMessage());
-                        return;
-                    }
+                    
+                    IAsyncEnumerable<ConfigurationClient> availableClients = _configClientManager.GetAvailableClients(cancellationToken);
+
+                    IAsyncEnumerator<ConfigurationClient> clientEnumerator = availableClients.GetAsyncEnumerator();
+                    await using (clientEnumerator.ConfigureAwait(false))

```

Comment:
`await using (clientEnumerator.ConfigureAwait(false))`

Why is the configure await needed if each individual call to MoveNextAsync is configured? Also, does ConfigureAwait return the original object? If not, I worry the using statement won't dispose `availableClients`

Reply:
This is from the .net offical doc, let me find it to you later

Reply:
The source `IAsyncDispoable` would be disposed
![image](https://github.com/Azure/AppConfiguration-DotnetProvider/assets/99175581/4ead4e03-b20e-4590-ac37-3a1129d30ac8)